### DISCUSSION TITLE
What are the characteristics of the server?

### DISCUSSION THREAD

**Discussion by dmitrymaslov99** [osm-search/Nominatim#2196] [Q&A]
Please tell me which server you need to buy, what to serve so many requests?
![image](https://user-images.githubusercontent.com/36110676/109809840-d39a0a80-7c39-11eb-9ce9-a844771fc9e3.png)
RAM, processor, number of cores, etc.

**Comment by mtmail** [comment]
Moving the question to discussions/Q&A

**Comment by mtmail** [comment]
What data will you import? The whole world or one country (which country)?

Can you give 1-2 example searches? Different types of searches have different complexity on the database.

What do the columns in the image represent?

**Reply by dmitrymaslov99** [reply]
1. So far only Russia
2. Geocoding and reverse geocoding will be used (https://nominatim.org/release-docs/develop/api/Search/) (https://nominatim.org/release-docs/develop/api/Reverse/)
3. The number of requests per second. About 110 requests per second and it slows down. That is, the server must be able to withstand a load of about 150 requests per second. (About 7500-10000 per minute)
4. 
> Какие данные вы будете импортировать? Весь мир или одна страна (какая страна)?
> 
> Вы можете привести 1-2 примера поиска? Различные типы поиска имеют разную сложность в базе данных.
> 
> Что представляют собой столбцы на изображении?

**Comment by dmitrymaslov99** [comment]
1. So far only Russia
2. Geocoding and reverse geocoding will be used (https://nominatim.org/release-docs/develop/api/Search/) (https://nominatim.org/release-docs/develop/api/Reverse/)
3. The number of requests per second. About 110 requests per second and it slows down. That is, the server must be able to withstand a load of about 150 requests per second. (About 7500-10000 per minute)

**Reply by mtmail** [reply]
https://nominatim.org/release-docs/latest/admin/Installation/#hardware

Russia raw *.pbf data is 3GB compared to 60GB for the whole planet, so you should be fine with 100GB disc space. Best if it's fast (NVME drives).

Import will work with 16GB RAM, 32GB RAM should fit the most used database indices and is better for query performance, using 64GB will be even better, more RAM will likely have no additional benefit.

I think you might be good enough with 12 CPU cores. Some will be used by the database constantly querying, sorting, others will be multithreaded PHP handling the HTTP requests. You will need to setup php-fpm to handle the incoming requests (see https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-20/) and then finetune its configuration for the number of CPU cores and level of multithreading. Some CPUs are better at this than others.

If you have the time try a setup at one of the cloud providers (pay per hour) first where you're more flexible with adding CPU core or disk space before purchasing hardware.

Hardware recommendations are not an exact science, there's too much different hardware available.
